### PR TITLE
feat: support Azure Monitor datasource

### LINKED ISSUE #345: Azure Monitor Datasource Support
Grafana 8.0: https://grafana.com/docs/grafana/latest/datasources/azuremonitor/

### PR DESCRIPTION
**Pull request by aymericDD**
# What

Add Azure Monithor datasource.

# Why

In the recent versions of Grafana, the Azure Monitor has been added. The purpose of this PR is to support this new datasource.

Fixes and closes #345

### CODE REVIEW

File: grafana/resource_data_source.go [hunk 1]
```diff
@@ -125,6 +125,11 @@ source selected (via the 'type' argument).
 							Optional:    true,
 							Description: "(Stackdriver) The authentication type: `jwt` or `gce`.",
 						},
+						"azure_log_analytics_same_as": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Description: "(Azure Monitor) The log analytics same as.",
+						},
```
Comment (julienduchesne) [inline grafana/resource_data_source.go]: I just saw that this one is deprecated now. https://github.com/grafana/grafana/blob/23956557d8c6a119b7de5be5c42024e29634d002/pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go#L150-L153. We can leave it out

File: grafana/resource_data_source.go [hunk 2]
```diff
@@ -237,6 +252,21 @@ source selected (via the 'type' argument).
...
+						"log_analytics_default_workspace": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "(Azure Monitor) Log analytics default workspace.",
+						},
+						"log_analytics_tenant_id": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "(Azure Monitor) Log analytics tenant ID.",
+						},
```
Comment (julienduchesne) [inline grafana/resource_data_source.go]: Seems like all log analytics values are deprecated

https://github.com/grafana/grafana/blob/d5f40b63bbafe0126086d42bde2caae59bed452b/public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts#L228-L234

File: grafana/resource_data_source.go (hunk 1 above)
Comment (aymericDD) [reply grafana/resource_data_source.go]: Sorry :pray:  This field is removed

File: grafana/resource_data_source.go (hunk 2 above)
Comment (aymericDD) [reply grafana/resource_data_source.go]: Alright, All fields of Azure Monitor with the log_analytics prefix are removed :+1:

**Comment by inkel** [comment]
@aymericDD thanks for your contribution! It seems some tests are failing. I just pushed an update to `master` to [better identify failing tests](https://github.com/grafana/terraform-provider-grafana/commit/9161afc024f6eaa87b6604eeb38c98251878d0f9?w=1), can you rebase your branch and push again?

File: grafana/resource_data_source.go [hunk 3]
```diff
@@ -332,6 +342,16 @@ source selected (via the 'type' argument).
 							Optional:    true,
 							Description: "(PostgreSQL) SSLmode. 'disable', 'require', 'verify-ca' or 'verify-full'.",
 						},
+						"subscription_id": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "(Azure Monitor) The subscription id",
+						},
+						"tenant_id": {
+							Type:        schema.TypeBool,
```
Comment (julienduchesne) [inline grafana/resource_data_source.go]:
```suggestion
							Type:        schema.TypeString,
```

Comment (aymericDD) [reply grafana/resource_data_source.go]: Fixed :yum:

**Comment by aymericDD** [comment]
All issues are fixed and the PR is rebase from the master. Sorry for the inconveniences. I don't find a way to run all tests with my local environment even with the **testacc-docker**. :disappointed:

**Review by julienduchesne — APPROVED** [review]
LGTM!

**Comment by aymericDD** [comment]
Any news about the release to allow the usage of this feature? @julienduchesne @inkel

**Comment by dohnto** [comment]
@aymericDD This feature is now released as part of https://github.com/grafana/terraform-provider-grafana/releases/tag/v1.21.2

**Comment by aymericDD** [comment]
Thanks! :+1:

**Pull request merged**